web/components/dialogs/FcDialogReportParameters.vue

```diff
@@ -0,0 +1,68 @@
+<template>
+  <v-dialog
+    v-model="internalValue"
+    max-width="300"
+    scrollable>
+    <v-card>
+      <v-card-title>Set Parameters</v-card-title>
+      <v-divider></v-divider>
+      <v-card-text>
+        <component
+          :is="'FcReportParameters' + reportType.suffix"
+          v-model="internalReportParameters"
+          @update-report="$emit('update-report')" />
+      </v-card-text>
+      <v-card-actions>
+        <v-spacer></v-spacer>
+        <v-btn
+          color="primary"
+          text
+          @click="internalValue = false">Cancel</v-btn>
+        <v-btn
+          color="primary"
+          text
+          @click="onClickSave">Save</v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script>
+import { ReportType } from '@/lib/Constants';
+import FcReportParametersWarrantTrafficSignalControl
+  from '@/web/components/reports/FcReportParametersWarrantTrafficSignalControl.vue';
+
+export default {
+  name: 'FcDialogReportParameters',
+  components: {
+    FcReportParametersWarrantTrafficSignalControl,
+  },
+  props: {
+    reportParameters: Object,
+    reportType: ReportType,
+    value: Boolean,
+  },
+  data() {
+    const internalReportParameters = JSON.parse(JSON.stringify(this.reportParameters));
+    return {
+      internalReportParameters,
+    };
+  },
+  computed: {
+    internalValue: {
```

Comment:
i'd originally tried to extract this `value` / `internalValue` pattern into a Vue mixin, but hit a stumbling block: `props` need types.  (well, they don't _need_ them, but the extra type-checking is good!)

there *is* a solution, however!  we can create a "mixin factory": a module that exports a function `getMixin(type)`, which itself returns a mixin.  we can then call `getMixin` within the `mixins` array to create the mixin instance, since under the hood it's just doing fancy object merging.  (no idea what this pattern would look like under the upcoming Vue 3.x composition API, but we don't need to worry about that - they've promised backwards compatibility for this major version at least.)

we should probably do this in a separate "cleanup" PR.